### PR TITLE
New ldoc themes

### PR DESCRIPTION
**Pull request by raven2cz**
# Draft: New ldoc themes

Add several themes according to discussion. Dynamic change. Support dark/light mode.

Open topics:
* @Elv13 svg images will need more our work. Maybe some cooperation here. Light/dark color is not integrated correctly to svg text and fill strokes. We have to find some solution.
* @actionless next topic for you about "look and feel" feelings and typography improvements. In common, there can be some css styling bugs. The documentation is huge, I don't try any page. This is just draft.

Demo is here:
https://raven2cz.github.io/awesome/

### CODE REVIEW

**Review by actionless — APPROVED** [review]
i think the current version is good for merging - in case of other potential improvements it might be better to roll a separate PR

**Comment by Elv13** [comment]
Nice work!

@actionless I think we should wait to merge it until the svg issues are resolved, right now it look like this:

Screenshot_20221007_232958

With high contrast enabled, it looks like this

Screenshot_20221007_233054

So it *can* work. Since this is a little problematic regression for some users (especially on bad TFT screens, where contrast isn't good even in idea conditions), I suggest this gets fixed first. After that, yeah, we can merge it! After that I guess we can iterate on the default dark theme. 

@raven2cz what kind of display are you using? The default theme looks fine on my LCD, but a little off on my OLED (which is oversaturated, smartphone-like). In any case, those are small details, I am super grateful of all the groundwork you did here! Exited to merge this soon.

**Comment by raven2cz** [comment]
* **Add Contrast and good visibility**
> what kind of display are you using? The default theme looks fine on my LCD, but a little off on my OLED (which is oversaturated, smartphone-like). In any case, those are small details, I am super grateful of all the groundwork you did here! Exited to merge this soon.

It is a reason why I used more themes, not just light and dark, because different contrast. Themes have almost **exactly defined colorschemes**, which cannot be changed. But some themes are good for lower contrast, some for middle, etc. 

For "blind" people is necessary to still use light mode. Awesome-dark is for "glowing" monitors. The one dark and monokai are for middle and low contrasts. From my side. Mainly I'm using monokai and sometimes one dark, because I have middle and low contrasts set. But I don't prefer light mode.

* **SVG Images Light/Dark themes**
The problem is that `currentcolor` in autogen svg images doesn't support dynamic change of css variables. I don't know why, but all my actual tests failed. Just a direct usage of color works, but it is not for us. In addition, my tests just change the text color, not other colors of fill/stroke. I don't find correct wat with `currentcolor` feature. If anybody here has some idea, please provide me some example of usage.

Next problem, that there are still images which are inserted by `img` element and `src=*.svg`.

But I've got a idea. Fast, simple and effective without huge changes, generation of images, etc. I add revert filter and xpath to apply to any img-object and svg files. In actual case, I revert 100% of colors from original images which are designed for light mode. I'm happy with it. I haven't better solution in common scope now. But for first merge, it is not bad, not bad.

Check it in demo 
https://raven2cz.github.io/awesome/core_components/tag.html

**Comment by actionless** [comment]
> The problem is that currentcolor

as far as i know, in modern browsers you can style with CSS the paths inside SVG almost same as in HTML

**Comment by raven2cz** [comment]
> > The problem is that currentcolor
> 
> as far as i know, in modern browsers you can style with CSS the paths inside SVG almost same as in HTML

yes, I was not successful with it for awesome exported images. But the described filter in prev post works relatively great. There are around 500 images in doc.

**Review by Elv13 — DISMISSED** [review]

**Comment by Elv13** [comment]
@raven2cz Is there anything else you want to do before we merge this? With the new filter, the dark theme is very readable. It would be nice to "really" implement all colors, but i think we can iterate for that?

**Comment by raven2cz** [comment]
> @raven2cz Is there anything else you want to do before we merge this? With the new filter, the dark theme is very readable. It would be nice to "really" implement all colors, but i think we can iterate for that?

yes, this first iteration is done. Yesterday, I go through all pages, fast check all and it is ok from my side. Mapping of image colors is next iteration cycle.

**Comment by raven2cz** [comment]
@actionless @Aire-One There is missing second approval. @Elv13 cannot merge this themes without it. Thanks for coop.

**Comment by Elv13** [comment]
I can merge it, but I need to override the checks. If there is no comment by Friday, I will.

File: docs/ldoc.css
Comment (Aire-One) [inline docs/ldoc.css:1214]: Should we remove these comment blocks?

**Review by Aire-One — APPROVED** [review]

**Comment by Aire-One** [comment]
> @actionless @Aire-One There is missing second approval. @Elv13 cannot merge these themes without it. Thanks for coop.

Sorry for the delay @raven2cz. I'm a little out of the loop here, but I think it's fine to merge these changes.

Thank you a lot for all the effort you're putting on the website/apidoc redesign. :rocket:

File: docs/ldoc.css
Comment (raven2cz) [reply docs/ldoc.css]: Done. However, it will now require a new review release from your side...

**Review by Aire-One — APPROVED** [review]

**Review by Elv13 — APPROVED** [review]

**Comment by actionless** [comment]
thanks a lot @raven2cz i wasn't reviewing it during wip, but i quickly went through end result - looking really slick 🔥

**Pull request merged**
